subprojects/core/src/main/java/org/gradle/api/internal/changedetection/changes/ShortCircuitTaskArtifactStateRepository.java

```diff
@@ -132,15 +127,20 @@ public void afterOutputsRemovedBeforeTask() {
             delegate.afterOutputsRemovedBeforeTask();
         }
 
+        @Override
+        public ImmutableSortedMap<String, CurrentFileCollectionFingerprint> snapshotAfterTaskExecution(TaskExecutionContext taskExecutionContext) {
+            return delegate.snapshotAfterTaskExecution(taskExecutionContext);
+        }
 
         @Override
-        public void snapshotAfterTaskExecution(Throwable failure, UniqueId buildInvocationId, TaskExecutionContext taskExecutionContext) {
-            delegate.snapshotAfterTaskExecution(failure, buildInvocationId, taskExecutionContext);
+        public void persistNewOutputs(ImmutableSortedMap<String, CurrentFileCollectionFingerprint> newOutputFingerprints, boolean successful, OriginMetadata originMetadata) {
+            delegate.persistNewOutputs(newOutputFingerprints, successful, originMetadata);
         }
 
+        @Nullable
         @Override
-        public void snapshotAfterLoadedFromCache(ImmutableSortedMap<String, CurrentFileCollectionFingerprint> newOutputFingerprints, OriginMetadata originMetadata) {
-            delegate.snapshotAfterLoadedFromCache(newOutputFingerprints, originMetadata);
+        public OverlappingOutputs getOverlappingOutputs() {
+            return null;
```

Comment:
I don't understand why we can avoid calculating overlapping outputs in `RerunTaskArtifactState`. We could just use `delegate.getOverlappingOutputs()`.

It seems a little bit strange returning null here. Can we move this state maybe somewhere else?

Reply:
This is an error. I wonder how no test caught this.